### ISSUE TITLE
Enable integer bitwise ops used by BERT token_type_ids decode (&, >>, bitwise_and_)

### ISSUE THREAD

**Issue by GOavi101** [torch-spyre/torch-spyre#3509]
### What feature would you like to be added?
Support the integer bitwise ops used by vLLM's bert._decode_token_type_ids on Spyre (int32/int64), roughly:

```
  ids_mask = 1 << TOKEN_TYPE_SHIFT
  token_type_ids = (input_ids & ids_mask) >> TOKEN_TYPE_SHIFT
  input_ids.bitwise_and_(~ids_mask)
```

Ops involved: left/right shift, bitwise and, bitwise and_ (in-place), and bitwise not / invert on integer tensors.

### Why is this feature needed?
spyre-inference currently stubs bert._decode_token_type_ids to torch.zeros_like(input_ids) because Spyre cannot lower this unpack. That forces segment 0 only.

Single-segment Granite embeddings are fine; multi-segment / true token_type_ids BERT inputs are wrong until these ops work on Spyre (or we keep a permanent CPU decode, which still needs a correct host path instead of zeros).

**Comment by mudhakar** [comment]
This should be solved with an adapter, not by enabling the ops.

Bit packing is not a BERT requirement...  it is a vLLM transport hack, per comment in vllm/model_executor/models/bert.py...

Adapters: 
1. Intercept _encode_token_type_ids --> no-op the packing; instead copy_ the real token_type_ids into a persistent device buffer (zero-filled past token_type_ids.shape[0], which is the correct semantics for right-padding anyway... pad tokens are segment 0).
2. Intercept _decode_token_type_ids --> return that buffer, and don't touch input_ids.

**Comment by GOavi101** [comment]
Thank you for the suggestion. I’ve implemented the adapter approach here: https://github.com/torch-spyre/spyre-inference/pull/481
